src/UAEPass/Provider.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace SocialiteProviders\UAEPass;
+
+use GuzzleHttp\RequestOptions;
+use Illuminate\Support\Arr;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
+
+/**
+ * @see https://docs.uaepass.ae/guides/authentication/web-application
+ */
+class Provider extends AbstractProvider
+{
+    public const IDENTIFIER = 'UAEPASS';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = ['urn:uae:digitalid:profile:general'];
+
+    protected $scopeSeparator = ':';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase(
+            $this->getBaseUrl().'/idshub/authorize',
+            $state
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return $this->getBaseUrl().'/idshub/token';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        $response = $this->getHttpClient()->get(
+            $this->getBaseUrl().'/idshub/userinfo',
+            [
+                RequestOptions::HEADERS => [
+                    'Authorization' => 'Bearer '.$token,
+                ],
+            ]
+        );
+
+        return json_decode((string) $response->getBody(), true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function mapUserToObject(array $user)
+    {
+        return (new User())->setRaw($user)->map([
+            'uuid'          => Arr::get($user, 'uuid'),
+            'sub'           => Arr::get($user, 'sub'),
+            'fullnameAR'    => Arr::get($user, 'fullnameAR'),
+            'firstnameAR'   => Arr::get($user, 'firstnameAR'),
+            'lastnameAR'    => Arr::get($user, 'lastnameAR'),
+            'firstnameEN'   => Arr::get($user, 'firstnameEN'),
+            'lastnameEN'    => Arr::get($user, 'lastnameEN'),
+            'fullnameEN'    => Arr::get($user, 'fullnameEN'),
+            'profileType'   =>  Arr::get($user, 'profileType'),
+            'unifiedID'     => Arr::get($user, 'unifiedID'),
+            'email'         => Arr::get($user, 'email'),
+            'idn'           => Arr::get($user, 'idn'),
+            'gender'        => Arr::get($user, 'gender'),
+            'mobile'        => Arr::get($user, 'mobile'),
+            'userType'      => Arr::get($user, 'userType'),
+            'nationalityEN' => Arr::get($user, 'nationalityEN'),
+            'nationalityAR' => Arr::get($user, 'nationalityAR'),
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAccessTokenResponse($code)
+    {
+        $response = $this->getHttpClient()->post($this->getTokenUrl(), [
+            RequestOptions::HEADERS => [
+                'Authorization' => 'Basic '.base64_encode($this->clientId.':'.$this->clientSecret),
+                'Content-Type'  => 'multipart/form-data',
```

Comment:
why is this a multipart request? there's no body being sent

Reply:
While invoking getAccessTokenResponse method, it's mandatory to send **Content-Type with multipart/form-data**, otherwise it's throwing 500 error.

For reference, please check the below link
[UAEPass official Integration Document](https://docs.uaepass.ae/guides/authentication/web-application/2.-obtaining-the-access-token#curl-request-for-token-generation-call)